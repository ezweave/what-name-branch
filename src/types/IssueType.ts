export enum IssueType {
  BUG = 'bug',
  SPIKE = 'spike',
  STORY = 'story',
}
