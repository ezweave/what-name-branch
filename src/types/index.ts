export * from './Issue';
export * from './IssueType';
