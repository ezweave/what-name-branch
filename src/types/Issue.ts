import { IssueType } from './IssueType';

export interface Issue {
  description: string,
  name: string,
  type: IssueType,
}
