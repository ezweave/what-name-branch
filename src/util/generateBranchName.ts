import { toUpper } from 'lodash';
import { Issue } from '@ezweave/types';
import { convertDescription } from './convertDescription';

interface GenerateBranchName {
  (issue: Issue): string,
}
 
export const generateBranchName: GenerateBranchName = ({
  description,
  name,
  type,
}) => `${type}/${toUpper(name)}/${convertDescription(description)}`;
