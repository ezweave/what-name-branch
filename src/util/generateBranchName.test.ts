import { IssueType } from '@ezweave/types';
import { generateBranchName } from './generateBranchName';

describe(generateBranchName, () => {
  it('generates a branch name, given proper inputs', () => {
    expect(
      generateBranchName({
        description: 'We\'ve got this terrible defect',
        name: 'MYPROJECT-1',
        type: IssueType.BUG,
      }),
    ).toMatchSnapshot();
  });
});
