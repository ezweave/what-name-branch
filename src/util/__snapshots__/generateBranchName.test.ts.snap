// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateBranchName generates a branch name, given proper inputs 1`] = `"bug/MYPROJECT-1/weve-got-this-terrible-defect"`;
